### ISSUE TITLE
Refactor Tor implementation to retain consistent profile

### ISSUE THREAD

**Issue by tomlowenthal** [brave/brave-browser#2307]
Because private windows with Tor are implemented using guest windows, they don't share anything with the regular profile. Settings aren't inherited, bookmarks aren't present, extensions aren't available, and so on. Instead, windows with Tor should match regular private windows in their interactions with the profile — generally read-only unless someone makes an affirmative action like adding a bookmark.

**Comment by yrliou** [comment]
@durch Hi, just a heads up that I am actively working on refactoring our tor profile stuff to make it align with the incognito profile (profile behind private window) at the moment. #1677 should be unblocked after this is done, thanks!

**Comment by durch** [comment]
Glad to hear that, is there a branch, or a PR I could follow? Just trying to get some ideas around the timeline.

**Comment by yrliou** [comment]
@durch At the moment it's still a working in progress in the local, but I will update here if a remote branch or PR is created for you and anyone who is interested to follow the progress, thanks!

**Comment by durch** [comment]
That sounds awesome @yrliou, thank you!

**Comment by yrliou** [comment]
With https://github.com/brave/brave-core/pull/3319 merged, we now have settings and bookmarks from the original regular profile, extensions are not ready yet, I'll open a new issue to capture that.

**Comment by btlechowski** [comment]
Verification passed on

Brave | 0.72.110 Chromium: 78.0.3904.70 (Official Build) dev (64-bit)
-- | --
Revision | edb9c9f3de0247fd912a77b7f6cae7447f6d3ad5-refs/branch-heads/3904@{#800}
OS | Ubuntu 18.04 LTS

Verified test plan from https://github.com/brave/brave-core/pull/3319
Verified bookmarks and global settings are shared between normal and tor window. Extensions are not shared yet per https://github.com/brave/brave-browser/issues/2307#issuecomment-538018323
Verified that brave://settings page is redirected to normal window.
